packages/langium/test/grammar/grammar-util.test.ts

```diff
@@ -86,6 +87,12 @@ describe('Find name assignment in parser rules', () => {
         expect(nameAssignment).toBe(undefined);
     });
 
+    test('Should be able to infer data type rule for cross ref', () => {
+        const a = rules[3];
+        const nameAssigment = findNameAssignment(a);
+        expect(nameAssigment?.feature).toBe('name');
+    });
+
 });
 
```

Comment:
I'd propose to add tests of `getTypeName` in `grammar-util.test.ts` too that it would have unit-tests. Then you can add "integration" tests in `inferred-types.spec.ts` + `type-validator.test.ts`.